bittensor/metagraph.py

```diff
@@ -291,6 +366,20 @@ def uids_to_neurons(self, uids: torch.Tensor) -> List[bittensor_pb2.Neuron]:
             response.append(self.state.neurons[idx])
         return response
 
+    def neurons_to_uids(self, neurons: List[bittensor_pb2.Neuron]) -> torch.LongTensor:
+        r""" Returns uids associated with the passed neurons.
+        Args:
+            neurons: (List[bittensor_pb2.Neuron]): 
+                neuron info ordered by passed uids.
+        Returns:
+            uids: (torch.LongTensor)
+                uids associated with neurons.
+        """
+        uids = []
+        for n in neurons:
+            uids.append(self.state.uid_for_pubkey[n.public_key])
+        return torch.tensor(uids)
+
```

Comment:
Can probably make this more pythonic: `return [self.state.uid_for_pubkey[n.public_key] for n in neurons]`